### PR TITLE
Typos in docs/sfm.rst

### PR DESCRIPTION
**Pull request by stoyanovd**
Hello, Chris!

I fix some typos in `sfm.rst` (with spell checking utility help).
And add few direct links to code.

There were some not apparent to me places:

1. In function `Reconstruction::AddTrack`:
     ViewId as argument, but in description "View name (i.e. string)"
    (maybe I misunderstood?)

2. `WilsonECCV2012` citation - There is `WilsonECCV2014` in the bibliography, so (I can't google `WilsonECCV2012` together with 1DSfM) I changed to 2014.

### CODE REVIEW

File: docs/source/sfm.rst [hunk 1]
```diff
@@ -53,13 +53,13 @@ Views and Tracks
...
 about features from the image, camera pose information, and EXIF
-information. Views make up our basic visiblity constraints and are a fundamental
+information. Views make up our basic visibility constraints and are a fundamental
 part of the SfM pipeline.
 
 .. class:: Track
 
 A :class:`Track` represents a feature that has been matched over potentially
-many images. When a feature appears in multiple images it typically means that
+many images. When a feature appears in multiple images, it typically means that
```
Comment (sweeneychris) [inline docs/source/sfm.rst]: No comma should go here. Commas should separate complete sentences/thoughts.

File: docs/source/sfm.rst [hunk 2]
```diff
@@ -817,9 +822,9 @@ Incremental SfM is generally considered to be more robust than global SfM
 methods; however, it requires many more instances of bundle adjustment (which
 is very costly) and so incremental SfM is not as efficient or scalable.
 
-To use the Incremental SfM pipeline, simple set the
+To use the Incremental SfM pipeline, simply set the
 ``reconstruction_estimator_type`` to
-``ReconstructionEstimatorType::INCREMENTAL``. There are many more options that
+``ReconstructionEstimatorType::INCREMENTAL``. There are much more options that
```
Comment (sweeneychris) [inline docs/source/sfm.rst]: It should stay as "many". The difference between "much" and "many" is that "much" refers to uncountable nouns and "many" refers to countable nouns. In this case, we can count the number of options so it should be "many".

File: docs/source/sfm.rst [hunk 3]
```diff
@@ -1371,7 +1378,7 @@ Similarity Transformation
 
   .. function:: void AlignPointCloudsUmeyama(const int num_points, const double left[], const double right[], double rotation[], double translation[], double* scale)
 
-    This function estimates the 3D similiarty transformation using the least
+    This function estimates the 3D similarty transformation using the least
```
Comment (sweeneychris) [inline docs/source/sfm.rst]: similarity  (note the "i" after the "r")

**Comment by sweeneychris** [comment]
Thanks for doing this! Just a few quick changes then I'll merge this into the main repo.

**Comment by stoyanovd** [comment]
Not simple cases for me, thank you for explanations =)

By the way, I found some tools for sphinx - maybe it will be useful:
https://docs.djangoproject.com/en/1.9/internals/contributing/writing-documentation/#spelling-check
If it won't cause more problems than good...

**Pull request merged**